txt/src/carouselView/dev/Shared.iOS/CarouselViewRenderer.cs

```diff
@@ -30,463 +30,473 @@
+        }
+        void OnCollectionChanged(object source, NotifyCollectionChangedEventArgs e)
+        {
+            switch (e.Action)
+            {
+                case NotifyCollectionChangedAction.Add:
+                    _controller.ReloadData();
+
+                    if (e.NewStartingIndex <= _position)
+                        ShiftPosition(e.NewItems.Count);
+
+                    break;
+
+                case NotifyCollectionChangedAction.Move:
+                    for (var i = 0; i < e.NewItems.Count; i++)
+                    {
+                        _controller.MoveItem(
+                            oldPosition: e.OldStartingIndex + i,
+                            newPosition: e.NewStartingIndex + i
+                        );
+                    }
+                    break;
+
+                case NotifyCollectionChangedAction.Remove:
+                    if (Controller.Count == 0)
+                        throw new InvalidOperationException("CarouselView must retain a least one item.");
+
+                    var removedPosition = e.OldStartingIndex;
+
+                    if (removedPosition == _position)
+                    {
+                        _controller.DeleteItems(
+                            Enumerable.Range(e.OldStartingIndex, e.OldItems.Count)
+                        );
+                        if (_position == Controller.Count)
+                            OnPositionChange(_position - 1);
+                        OnItemChange(_position);
+                    }
+
+                    else if (removedPosition > _position)
+                    {
+                        _controller.DeleteItems(
+                            Enumerable.Range(e.OldStartingIndex, e.OldItems.Count)
+                        );
+                    }
+
+                    else
+                        ShiftPosition(-e.OldItems.Count);
+
+                    break;
+
+                case NotifyCollectionChangedAction.Replace:
+                    _controller.ReloadItems(
+                        Enumerable.Range(e.OldStartingIndex, e.OldItems.Count)
+                    );
+                    break;
+
+                case NotifyCollectionChangedAction.Reset:
+                    _controller.ReloadData();
+                    break;
+
+                default:
+                    throw new Exception();
+            }
+        }
+        void ShiftPosition(int offset)
+        {
+            // By default the position remains the same which causes an animation in the case
+            // of the added/removed position preceding the current position. I prefer the constructed
+            // Android behavior whereby the item remains the same and the position changes.
+            var position = _position + offset;
+            _controller.ReloadData(position);
+            OnPositionChange(position);
+        }
+
+        protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
```

Comment:
It seens that visual studio changed the spacing from the lines, i only changed this block of code.